### PR TITLE
Remove sget command

### PR DESCRIPTION
**Pull request by luhring**
#### Summary

Removes the `sget` command for now, while the team refines the vision for the "sget" idea, how it will be manifested in code, and how we'll distribute it to users.

#### Ticket Link

Related to #1363 

#### Release Note

```release-note
Removes sget command
```

### CODE REVIEW

**Review by cpanato — CHANGES_REQUESTED** [review]
also need to remove from the goreleaser:
https://github.com/sigstore/cosign/blob/main/.goreleaser.yml#L147

and from build image using ko
https://github.com/sigstore/cosign/blob/main/Makefile#L140
https://github.com/sigstore/cosign/blob/main/release/release.mk#L22

and general build
https://github.com/sigstore/cosign/blob/main/Makefile#L92

**Comment by luhring** [comment]
Thanks @cpanato for finding those! 🙏 

I also found a handful of other spots we're using the `sget` command, and I removed them. Let me know if this was too heavy-handed.

Also, I left the `sget` Go package for now, since it's used in `cosign policy init` and in E2E testing. Let me know if you think we should work to remove this "sget" reference as well.

**Review by cpanato — DISMISSED** [review]
LGTM

will defer to @dlorenc

**Comment by cpanato** [comment]
would merge this after the 1.5.1 release

**Comment by luhring** [comment]
> I'd honestly still rather wait here until we have a plan. I don't think it's going to take forever to get something drafted, but doing it in a rush before we've figured out what the next step is feels like a mistake.

Agreed 💯 . Let's do that, and this PR is here if/when we want it.

**Comment by dlorenc** [comment]
Ok so for a plan, we can leave this around until the new repo is created and we've done a release from there. That way we can delete here without causing too much confusion.

**Comment by jeff-mccoy** [comment]
Thanks @dlorenc...we already included the `sget` package in our go project not realizing it was possibly being removed 😳.  Thanks to @imjasonh for pointing this out to us in slack.

**Comment by imjasonh** [comment]
@dlorenc ping, possibly https://github.com/sigstore/cosign/pull/2019 first?

**Comment by imjasonh** [comment]
I think we can do this now.

#2019 was merged in https://github.com/sigstore/cosign/commit/1f42a247b80d34f1deecfdf61ce8b5e3a858308b and released as part of cosign 1.12. We're on 1.13 now, so I think this is ample warning for folks to stop depending on it.

WDYT @dlorenc

**Comment by cpanato** [comment]
@imjasonh, when we make the release of the sget in the other repo we will continue the release version of sget and do a  `v1.14.0` lets, say.

**Comment by cpanato** [comment]
Not stale

**Comment by cpanato** [comment]
@luhring @imjasonh @dlorenc any plans on this? should we move forward or close this PR?

**Comment by znewman01** [comment]
I think we should merge this PR. As-is, sget is only marginally useful. There's no real replacement but then again sget itself was never complete and often encouraged security antipatterns.

**Comment by luhring** [comment]
> I think we should merge this PR.

Just catching up here! 😅  I'm happy to go through the merge conflicts and resolve them if current maintainers are still into this change! Otherwise we can close this.

**Comment by priyawadhwa** [comment]
Removing it sgtm!

File: .goreleaser.yml
Comment (cpanato) [inline .goreleaser.yml:161]: from where this come from? this was deprecated a long time and not have it own repo (policy-controller)

**Comment by cpanato** [comment]
@luhring maybe you rebased using an older base?
it is okay to close this and I open a fresh one?

**Comment by luhring** [comment]
@cpanato Please! 🙏

**Comment by cpanato** [comment]
closing in favor of https://github.com/sigstore/cosign/pull/2885

**Pull request closed without merging**
